VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionCommentsAdapter.java

```diff
@@ -27,9 +29,9 @@
         void reportComment(DiscussionComment comment);
     }
 
-    @Inject
-    public DiscussionCommentsAdapter(Context context, Listener listener, DiscussionComment response) {
+    public DiscussionCommentsAdapter(Context context, Router router, Listener listener, DiscussionComment response) {
```

Comment:
Actually, the view should not even know about the Router, so I moved the routing logic out of the adapter.